src/Classes/PartyTab.lua

```diff
@@ -0,0 +1,620 @@
+			end
+			if node.attrib.name ~= "EnemyConditions" and node.attrib.name ~= "EnemyMods" then
+				self:ParseBuffs(self.buffExports, node.attrib.string, node.attrib.name)
+			end
+			--self:ParseBuffs(self.buffExports, node.attrib.string, "Aura")
+			--self:ParseBuffs(self.buffExports, node.attrib.string, "Curse")
+			--self:ParseBuffs(self.buffExports, node.attrib.string, "EnemyConditions")
+			--self:ParseBuffs(self.buffExports, node.attrib.string, "EnemyMods")
+		end
+	end
+	self.lastContentAura = self.controls.editAuras.buf
+	self.lastContentCurse = self.controls.editCurses.buf
+	self.lastContentEnemyCond = self.controls.enemyCond.buf
+	self.lastContentEnemyMods = self.controls.enemyMods.buf
+	self.lastEnableExportBuffs = self.enableExportBuffs
+end
+
+function PartyTabClass:Save(xml)
+	local child = { elem = "ImportedText", attrib = { name = "Aura" } }
+	child.attrib.string = self.controls.editAuras.buf
+	t_insert(xml, child)
+	child = { elem = "ImportedText", attrib = { name = "Curse" } }
+	child.attrib.string = self.controls.editCurses.buf
+	t_insert(xml, child)
+	child = { elem = "ImportedText", attrib = { name = "EnemyConditions" } }
+	child.attrib.string = self.controls.enemyCond.buf
+	t_insert(xml, child)
+	child = { elem = "ImportedText", attrib = { name = "EnemyMods" } }
+	child.attrib.string = self.controls.enemyMods.buf
+	t_insert(xml, child)
+	child = { elem = "ExportedBuffs", attrib = { name = "Aura" } }
+	child.attrib.string = self:exportBuffs("Aura")
+	t_insert(xml, child)
+	child = { elem = "ExportedBuffs", attrib = { name = "Curse" } }
+	child.attrib.string = self:exportBuffs("Curse")
+	t_insert(xml, child)
+	child = { elem = "ExportedBuffs", attrib = { name = "EnemyConditions" } }
+	child.attrib.string = self:exportBuffs("EnemyConditions")
+	t_insert(xml, child)
+	child = { elem = "ExportedBuffs", attrib = { name = "EnemyMods" } }
+	child.attrib.string = self:exportBuffs("EnemyMods")
+	t_insert(xml, child)
+	self.lastContentAura = self.controls.editAuras.buf
+	self.lastContentCurse = self.controls.editCurses.buf
+	self.lastContentEnemyCond = self.controls.enemyCond.buf
+	self.lastContentEnemyMods = self.controls.enemyMods.buf
+	self.lastEnableExportBuffs = self.enableExportBuffs
+end
+
+function PartyTabClass:Draw(viewPort, inputEvents)
+	self.x = viewPort.x
+	self.y = viewPort.y
+	self.width = viewPort.width
+	self.height = viewPort.height
+
+	for id, event in ipairs(inputEvents) do
+		if event.type == "KeyDown" then
+			if event.key == "z" and IsKeyDown("CTRL") then
+				if self.controls.editAuras.hasFocus then
+					self.controls.editAuras:Undo()
+				elseif self.controls.editCurses.hasFocus then
+					self.controls.editCurses:Undo()
+				end
+			elseif event.key == "y" and IsKeyDown("CTRL") then
+				if self.controls.editAuras.hasFocus then
+					self.controls.editAuras:Redo()
+				elseif self.controls.editCurses.hasFocus then
+					self.controls.editCurses:Redo()
+				end
+			end
+		end
+	end
```

Comment:
I'm surprised you have to explicitly call Undo/Redo for the inputs here.  I thought if you had them focused they'd automatically use those handlers.

Reply:
iirc those were copied from the notes tab, and I dont think they worked without me explicitly adding them here